extensions/amp-story/1.0/test/test-amp-story.js

```diff
@@ -1198,6 +1199,147 @@ describes.realWin(
         });
       });
 
+      describe('amp-story-subscriptions navigation', () => {
+        let subscriptionsEl;
+
+        beforeEach(async () => {
+          toggleExperiment(win, 'amp-story-subscriptions', true);
+
+          await createStoryWithPages(4, [
+            'cover',
+            'page-1',
+            'page-2',
+            'page-3',
+          ]);
+          subscriptionsEl = win.document.createElement(
+            'amp-story-subscriptions'
+          );
+          story.element.appendChild(subscriptionsEl);
+
+          // buildCallback() could be omitted as long as createStoryWithPages()
+          // is called because the getImpl() call inside it.
+          await story.layoutCallback();
+        });
+
+        it('should continue navigating to locked pages after the subscription state gets resolved to granted', async () => {
+          // Block access with pending subscriptions state.
+          await story.switchTo_('page-1');
+          story.switchTo_('page-2');
+          expect(story.activePage_.element.id).to.equal('page-1');
+
+          // Once grant status resolves to true, goes back to normal story experiences.
+          story.storeService_.dispatch(
+            Action.TOGGLE_SUBSCRIPTIONS_STATE,
+            SubscriptionsState.GRANTED
+          );
+          await nextTick();
+          expect(story.activePage_.element.id).to.equal('page-2');
```

Comment:
also I think we can get this from the store, in general lets try to avoid using any privates in tests if possible.

Reply:
Yeah agree that we should avoid privates if possible, but I think for activePage_ is would be unavoidable because active page is not recorded in the store?